api/src/commonMain/kotlin/dev/kord/cache/api/observables/DataCache.kt

```diff
@@ -0,0 +1,44 @@
+package dev.kord.cache.api.observables
+
+/**
+ * An implementation of [TypedCache] that stores a set of [EntryCache]s.
+ */
+public class DataCache : TypedCache {
+
+    /**
+     * The set of [EntryCache]s stored in this [TypedSetCache].
+     */
+    private val types: MutableSet<EntryCache<Any>> = mutableSetOf()
+
+    /**
+     * Returns the [EntryCache] for the specified type, or `null` if it is not found.
+     */
+    private suspend fun <T : Any> getTypeOrNull(): EntryCache<T>? {
+        return toSet().filterIsInstance<EntryCache<T>>().singleOrNull()
```

Comment:
this will not work as expected, `filterIsInstance` can't check for the generic type `T`

Reply:
You are right because of type erasure. 
It's an oversight on my end.